### PR TITLE
Add initial multicolor support for 3 color displays

### PR DESCRIPTION
**Pull request by SaberShip**
I've done a quick implementation of "Accent Color" support for the 3 color e-paper displays. 

Simply added a configuration for defining the accent color if DISP_3C is defined.
Updated display_utils to support setting color or using a default if not supplied. 
I've modified the following elements to use the defined accent color:
- The City/State
- Currently Active Weather Warnings
- Temperature Y-axis labels 
- Temperature Line Graph (Not the precip. chance bars)
- BME Warning in status bar (if present)
- WiFi signal status (if below configured threshold) 
- Battery status (if below configured threshold)
- All fullscreen error icons

In the future it could be relatively simple to add more configurable color thresholds for current conditions (like UV, AQI, Visibility, etc...)

Below is an image of a 3 color display running under normal conditions:
IMG_20230415_191125

If required, I can upload more images showing different conditions with accent colors.

### CODE REVIEW

**Comment by lmarzen** [comment]
@SaberShip I'll take a closer look this weekend if I have time and get back to you.

File: platformio/src/renderer.cpp [hunk 1]
```diff
@@ -917,46 +918,53 @@ void drawStatusBar(String statusStr, String refreshTimeStr, int rssi,
...
 {
   String dataStr;
+  uint16_t dataColor = GxEPD_BLACK;
   display.setFont(&FONT_6pt8b);
   int pos = DISP_WIDTH - 2;
   const int sp = 2;
 
   // battery
   int batPercent = calcBatPercent(batVoltage);
+  if (batPercent < calcBatPercent(BATTERY_WARN_VOLTAGE)) {
```
Comment (lmarzen) [inline platformio/src/renderer.cpp:928]: Simplify this to 
`if (batVoltage < BATTERY_WARN_VOLTAGE) {`

File: platformio/src/renderer.cpp [hunk 2]
```diff
@@ -917,46 +918,53 @@ void drawStatusBar(String statusStr, String refreshTimeStr, int rssi,
...
+  drawString(pos, DISP_HEIGHT - 1 - 2, dataStr, RIGHT, dataColor);
   pos -= getStringWidth(dataStr) + 25;
   display.drawInvertedBitmap(pos, DISP_HEIGHT - 1 - 17,
-                             getBatBitmap24(batPercent), 24, 24, GxEPD_BLACK);
+                             getBatBitmap24(batPercent), 24, 24, dataColor);
   pos -= sp + 9;
 
   // wifi
   dataStr = String(getWiFidesc(rssi));
+  dataColor = getWiFiColor(rssi, GxEPD_BLACK, ACCENT_COLOR);
```
Comment (lmarzen) [inline platformio/src/renderer.cpp:941]: Can we simplify this to something like this...
`dataColor = rssi >= -70 ? GxEPD_BLACK : ACCENT_COLOR'`

**Review by lmarzen — CHANGES_REQUESTED** [review]
Overall these additions look great, I just had two minor comments that must be addressed before merging.

**Comment by SaberShip** [comment]
Of course!
Done.

**Review by lmarzen — APPROVED** [review]
Approved.

**Comment by lmarzen** [comment]
Thank you for your contribution @SaberShip.

**Pull request merged**